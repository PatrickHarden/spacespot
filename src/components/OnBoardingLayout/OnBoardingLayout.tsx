import React from 'react'
import styled from 'styled-components'
import Header from '../MainLayout/Header'
import Toaster from '../MainLayout/Toaster'
import Footer from 'components/MainLayout/Footer'

const Layout = styled.div`
  display: flex;
  flex-direction: column;
`
type LayoutProps = {
  children: JSX.Element | null
  hasFooter?: boolean
}
const OnBoardingLayout = (props: LayoutProps) => (
  <Layout>
    <Header />
    <Toaster />
    {props.children}
    {props.hasFooter && <Footer />}
  </Layout>
)

export default OnBoardingLayout
