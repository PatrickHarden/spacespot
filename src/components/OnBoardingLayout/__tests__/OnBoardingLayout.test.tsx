import React from 'react'
import OnBoardingLayout from '../OnBoardingLayout'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntlCache: jest.fn(),
  createIntl: jest.fn(),
}))

it('renders without crashing', async () => {
  const wrapper = shallow(
    <OnBoardingLayout>
      <div></div>
    </OnBoardingLayout>,
  )
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})
