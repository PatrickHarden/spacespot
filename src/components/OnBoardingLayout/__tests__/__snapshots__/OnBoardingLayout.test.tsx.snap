// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.div>
  <Header />
  <Toaster />
  <div />
</styled.div>
`;
